rsciio/blockfile/_api.py

```diff
@@ -283,24 +276,28 @@ def file_reader(filename, lazy=False, mmap_mode=None, endianess="<"):
     if not lazy:
         f.seek(offset2)
         data = np.fromfile(f, dtype=endianess + "u1")
-    else:
-        data = np.memmap(f, mode=mmap_mode, offset=offset2, dtype=endianess + "u1")
-    try:
-        data = data.reshape((NY, NX, DP_SZ * DP_SZ + 6))
-    except ValueError:
-        warnings.warn(
-            "Blockfile header dimensions larger than file size! "
-            "Will attempt to load by zero padding incomplete frames."
-        )
+        try:
+            data = data.reshape((NY, NX, DP_SZ * DP_SZ + 6))
+        except ValueError:
+            warnings.warn(
+                "Blockfile header dimensions larger than file size! "
+                "Will attempt to load by zero padding incomplete frames."
+            )
         # Data is stored DP by DP:
         pw = [(0, NX * NY * (DP_SZ * DP_SZ + 6) - data.size)]
         data = np.pad(data, pw, mode="constant")
-        data = data.reshape((NY, NX, DP_SZ * DP_SZ + 6))
-
+        data = data.reshape((NY, NX, DP_SZ * DP_SZ + 6), order="C")
+    else:
+        data = memmap_distributed(
+            filename,
+            chunks=chunks,
+            offset=offset2,
+            shape=(NY, NX, DP_SZ * DP_SZ + 6),
+            dtype=np.dtype(endianess + "u1"),
```

Comment:
I think you should be able to use a custom defined numpy dtype here? 

Thats what the `key` parameter is used for. It just throws away the other keys any only returns the one key that is relevent. 

https://github.com/hyperspy/rosettasciio/blob/f94acf730bc5e5c5a76a7dc8374f284575c8d064/rsciio/utils/distributed.py#L115-L123

Reply:
You mean using a custom dtype instead of making the shape slightly larger and then cropping?

Reply:
Done.